include/deal.II/fe/fe_tools.h

```diff
@@ -841,148 +841,234 @@ namespace FETools
+   * weight factors), or in cases where the shape functions one
+   * combines are not linearly dependent.
+   *
+   * </ol>
    */
-  template <int dim, int spacedim>
-  FiniteElementData<dim>
-  multiply_dof_numbers (const std::vector<const FiniteElement<dim,spacedim>*> &fes,
-                        const std::vector<unsigned int>                       &multiplicities,
-                        const bool do_tensor_product = true);
+  namespace Compositing
+  {
 
-  /**
-   * Same as above but for a specific number of sub-elements.
-   */
-  template <int dim, int spacedim>
-  FiniteElementData<dim>
-  multiply_dof_numbers (const FiniteElement<dim,spacedim> *fe1,
-                        const unsigned int            N1,
-                        const FiniteElement<dim,spacedim> *fe2=NULL,
-                        const unsigned int            N2=0,
-                        const FiniteElement<dim,spacedim> *fe3=NULL,
-                        const unsigned int            N3=0,
-                        const FiniteElement<dim,spacedim> *fe4=NULL,
-                        const unsigned int            N4=0,
-                        const FiniteElement<dim,spacedim> *fe5=NULL,
-                        const unsigned int            N5=0);
+    /**
+     * Take vectors of finite elements and multiplicities and multiply out
+     * how many degrees of freedom the composed element has per vertex,
+     * line, etc.
+     *
+     * If @p do_tensor_product is true, the number of components
+     * returned in the FiniteElementData object is the sum over the
+     * product of the number of components in each of the finite
+     * elements times the corresponding multiplicity.  Otherwise the
+     * number of components is taken from the first finite element with
+     * non-zero multiplicity, and all other elements with non-zero
+     * multiplicities need to have the same number of vector components.
+     *
+     * See the documentation of namespace FETools::Compositing for more
+     * information about the @p do_tensor_product argument.
+     */
+    template <int dim, int spacedim>
+    FiniteElementData<dim>
+    multiply_dof_numbers (const std::vector<const FiniteElement<dim,spacedim>*> &fes,
+                          const std::vector<unsigned int>                       &multiplicities,
+                          const bool do_tensor_product = true);
 
-  /**
-   * Compute the "restriction is additive" flags (see the
-   * documentation of the FiniteElement class) for a list of finite
-   * elements with multiplicities given in the second argument.
-   */
-  template <int dim, int spacedim>
-  std::vector<bool>
-  compute_restriction_is_additive_flags (const std::vector<const FiniteElement<dim,spacedim>*> &fes,
-                                         const std::vector<unsigned int>              &multiplicities);
+    /**
+     * Same as above but for a specific number of sub-elements.
```

Comment:
and <code> do_tensor_product=true</code>.

p.s. since this function is used in FESystem only, i did not extend it to the second case. That would probably not make much sense given all the defaults.